<launch>

    <!-- PARAMS -->
    <arg name="beta" default="0.00349"/>
    <arg name="gamma" default="0.109"/>
    <arg name="mu" default="1.0"/>

    <!-- Trigger -->
    <arg name="compute_ls_trigger_taun_min" default="0.002"/>
    <arg name="compute_ls_trigger_taun_max" default="0.010"/>
    <arg name="compute_ls_trigger_taun_when_trigger_active" default="$(arg compute_ls_trigger_taun_min)"/>
    <arg name="compute_static_trigger_taun_min" default="0.002"/>
    <arg name="compute_static_trigger_taun_max" default="0.010"/>
    <arg name="compute_static_trigger_taun_when_trigger_active" default="0.0005"/>

    <!-- Cor_tilde -->
    <arg name="max_sigma" value="30.0"/>
    <arg name="max_cor_tilde" default="0.8"/>

    <!-- FindZero -> Fn_min -->
    <arg name="default_fn_min" default="20.0"/>
    <arg name="newton_gain" default="0.3"/>
    <arg name="newton_cost_tol" default="1.0E-6"/>
    <arg name="newton_lambda" default="1.0E-10"/>
    <arg name="newton_max_iter" default="1500"/>

    <!-- Secure vars -->
    <arg name="compute_ls_min_fn_min" default="0.0"/>
    <arg name="compute_static_secure_gain" default="1.2"/>
    <arg name="compute_static_min_fn_min" default="2.0"/>

    <!-- Sub -->
    <arg name="compute_ls_contact_force_topic" default="contact_force"/>
    <arg name="compute_static_contact_force_topic" default="$(arg compute_ls_contact_force_topic)"/>

    <!-- Pubs -->
    <arg name="cor_topic" default="cor"/>
    <arg name="min_force_topic" default="min_force"/>
    <arg name="max_force_topic" default="max_force"/>
    <arg name="generalized_force_topic" default="generalized_force"/>
    <arg name="static_force_topic" default="static_force"/>

    <!-- Srv -->
    <arg name="compute_static_pause_service" default="compute_static/pause"/>
	<arg name="compute_static_control_rotation_service" default="compute_static/control_rotation"/>

    
<!-- -->

	<node name="compute_ls" pkg="slipping_control" type="compute_ls" output="screen">

        <!-- PARAMS -->
        <param name="beta" value="$(arg beta)"/>
        <param name="gamma" value="$(arg gamma)"/>
        <param name="mu" value="$(arg mu)"/>

        <!-- Trigger -->
        <param name="trigger_taun_min" value="$(arg compute_ls_trigger_taun_min)"/>
        <param name="trigger_taun_max" value="$(arg compute_ls_trigger_taun_max)"/>
        <param name="trigger_taun_when_trigger_active" value="$(arg compute_ls_trigger_taun_when_trigger_active)"/>

        <!-- Cor_tilde -->
        <param name="max_sigma" value="$(arg max_sigma)"/>
        <param name="max_cor_tilde" value="$(arg max_cor_tilde)"/>

        <!-- FindZero -> Fn_min -->
        <param name="default_fn_min" value="$(arg default_fn_min)"/>
        <param name="newton_gain" value="$(arg newton_gain)"/>
        <param name="newton_cost_tol" value="$(arg newton_cost_tol)"/>
        <param name="newton_lambda" value="$(arg newton_lambda)"/>
        <param name="newton_max_iter" type="int" value="$(arg newton_max_iter)"/>

        <!-- Secure vars -->
        <param name="min_fn_min" value="$(arg compute_ls_min_fn_min)"/>

        <!-- Sub -->
        <param name="contact_force_topic" value="$(arg compute_ls_contact_force_topic)"/>

        <!-- Pubs -->
        <param name="cor_topic" value="$(arg cor_topic)"/>
        <param name="min_force_topic" value="$(arg min_force_topic)"/>
        <param name="max_force_topic" value="$(arg max_force_topic)"/>
        <param name="generalized_force_topic" value="$(arg generalized_force_topic)"/>

	</node>
    
<!-- --> 

    <node name="compute_static" pkg="slipping_control" type="compute_static" output="screen">

        <!-- PARAMS -->
        <param name="beta" value="$(arg beta)"/>
        <param name="gamma" value="$(arg gamma)"/>
        <param name="mu" value="$(arg mu)"/>

        <!-- Trigger -->
        <param name="trigger_taun_min" value="$(arg compute_static_trigger_taun_min)"/>
        <param name="trigger_taun_max" value="$(arg compute_static_trigger_taun_max)"/>
        <param name="trigger_taun_when_trigger_active" value="$(arg compute_static_trigger_taun_when_trigger_active)"/>

        <!-- FindZero -> Fn_min -->
        <param name="default_fn_min" value="$(arg default_fn_min)"/>
        <param name="newton_gain" value="$(arg newton_gain)"/>
        <param name="newton_cost_tol" value="$(arg newton_cost_tol)"/>
        <param name="newton_lambda" value="$(arg newton_lambda)"/>
        <param name="newton_max_iter" type="int" value="$(arg newton_max_iter)"/>

        <!-- Secure vars -->
        <param name="secure_gain" value="$(arg compute_static_secure_gain)"/>
        <param name="min_fn_min" value="$(arg compute_static_min_fn_min)"/>

        <!-- Sub -->
        <param name="in_topic" value="$(arg compute_static_contact_force_topic)"/>

        <!-- Pubs -->
        <param name="out_topic" value="$(arg static_force_topic)"/>

        <!-- Srv -->
        <param name="pause_service" value="$(arg compute_static_pause_service)"/>
		<param name="control_rotation_service" value="$(arg compute_static_control_rotation_service)"/>

	</node>

<!-- --> 

</launch>